app/services/supplementary_data.py

```diff
@@ -78,6 +91,24 @@ def get_supplementary_data(*, dataset_id: str, unit_id: str, survey_id: str) ->
     raise SupplementaryDataRequestFailed
 
 
+def decrypt_supplementary_data(supplementary_data: MutableMapping) -> Mapping:
+    if (encryption_key_id := supplementary_data.get("encryption_key_id")) and (
```

Comment:
Having a second look, the `kid` would be in the header (after deserialisation) therefore we don't actually need this property right?. If so, we should consider changing the approach to not include this in the response as it is redundant.

Can you confirm this. If so, I'll chat to the other about dropping this key.

Reply:
Confirmed can get the kid out of the jwe header so encryption_key_id can be dropped. 
Updated code and tests to reflect